pkg/descheduler/strategies/topologyspreadconstraint.go

```diff
@@ -140,11 +140,6 @@ func RemovePodsViolatingTopologySpreadConstraint(
 			// (this loop is where we count the number of pods per topologyValue that match this constraint's selector)
 			var sumPods float64
 			for i := range namespacePods.Items {
-				// skip pods that are being deleted.
-				if namespacePods.Items[i].DeletionTimestamp != nil {
```

Comment:
Removing this throws off the calculation of the topology size.

/cc @damemi to confirm.

Reply:
Correct this check needs to be here specifically so that pods with a DeletionTimestamp aren't included in the calculation of the topology size, see https://github.com/kubernetes-sigs/descheduler/issues/618

It's unfortunately a bit redundant to the eviction check but necessary